notebook/static/notebook/js/notebook.js

```diff
@@ -729,29 +714,23 @@ define(function (require) {
     };
 
     /**
-     * Get an array of the cells in the currently selected range
-     *
-     * @return {Array} The selected cells
-     */
-    Notebook.prototype.get_selected_cells = function () {
-        return this.get_cells().filter(function(cell) {
-            return cell.in_selection;
-        });
-    };
-
-    /**
-     * Get the indices of the currently selected range of cells.
+     * Extend the selected range
      *
-     * @return {Array} The selected cells' numeric indices
+     * @param {string} [direction='down'] - 'up' or 'down
      */
-    Notebook.prototype.get_selected_indices = function () {
-        var result = [];
-        this.get_cell_elements().filter(function (index) {
-            if ($(this).data("cell").in_selection === true) {
-                result.push(index);
-            }
-        });
-        return result;
+    Notebook.prototype.extend_marked = function(direction) {
+        
+        // Convert the direction to an index offset
+        var offset = direction === 'up' ? -1 : 1;
```

Comment:
After looking at this, I think that `extend_marked` should only accept integers and not `up/down`. Part of this is the difficulty in picking names like `up/down` and `above/below`.